packages/uikit/src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -39,21 +39,45 @@ const DropdownMenu: React.FC<React.PropsWithChildren<DropdownMenuProps>> = ({
   const isMenuShow = isOpen && ((isBottomNav && showItemsOnMobile) || !isBottomNav);
 
   useEffect(() => {
+    let timeoutId: NodeJS.Timeout | null = null;
+    let isMouseInDropdown = false;
+
     const showDropdownMenu = () => {
       setIsOpen(true);
     };
 
     const hideDropdownMenu = (evt: MouseEvent | TouchEvent) => {
       const target = evt.target as Node;
-      return target && !tooltipRef?.contains(target) && setIsOpen(false);
+      if (target && !tooltipRef?.contains(target) && isBottomNav && !isMouseInDropdown) {
```

Comment:
Actually there's a more elegant way to do this. We could use the `debounce` function from `lodash/debounce`, which comes with a `cancel` method.

We can turn the `hideDropdownMenu` function into a debounced version and call it on mouseleave then call the cancel function on mouseenter